cli/ts/commands/setVerifyingKeys.ts

```diff
@@ -49,8 +49,8 @@ export const setVerifyingKeys = async (
   if (subsidyZkeyPath && !existsSync(subsidyZkeyPath)) logError(`3 ${subsidyZkeyPath} does not exist.`);
 
   // extract the vks
-  const processVk: VerifyingKey = VerifyingKey.fromObj(await extractVk(processMessagesZkeyPath));
-  const tallyVk: VerifyingKey = VerifyingKey.fromObj(await extractVk(tallyVotesZkeyPath));
+  const processVk = VerifyingKey.fromObj((await extractVk(processMessagesZkeyPath)) as unknown as IVkObjectParams);
+  const tallyVk = VerifyingKey.fromObj((await extractVk(tallyVotesZkeyPath)) as unknown as IVkObjectParams);
```

Comment:
I think it's better to add type for `extractVk` function instead of casting it.

Reply:
What do you suggest? given we're using snarkjs to extract the vkey, I'm using a type which fits the output of snarkjs' function, which is also what snarkjs expects as input to the verification function. For instance in genProof, I don't need to cast it, but in setVerifyingKey I need to..

Reply:
You can add return type to the method so there will be only one cast:

```
const extractVk = async (zkeyPath: string): Promise<IVkObjectParams> => {
  const vk = await zKey.exportVerificationKey(zkeyPath);
  await cleanThreads();
  return vk as IVkObjectParams;
};
```

Reply:
So that way I would have lost some properties that snarkjs needs for the verify function, so ended up changing the type in domainobjs.. pushing in a sec

Reply:
@0xmad fixed, please take a look! also updated the diagram, hopefully this time it renders properly for you